src/test/java/org/snomed/snowstorm/core/data/services/ConceptServiceTest.java

```diff
@@ -1699,6 +1703,138 @@ private boolean waitUntil(Supplier<Boolean> supplier, int maxSecondsToWait) {
+		desc.setAssociationTargets(Maps.newHashMap("POSSIBLY_EQUIVALENT_TO", Sets.newHashSet(anotherDescriptionId)));
+		Concept transferredConcept = simulateRestTransfer(concept);
+		conceptService.update(transferredConcept, mainBranchPath);
+		concept = conceptService.find(concept.getConceptId(), mainBranchPath);
+
+		// Create next version
+		codeSystemService.createVersion(codeSystem, 20210331, "20210331");
+
+		// Make the concept inactive
+		concept.setActive(false);
+		concept.setInactivationIndicator(Concepts.inactivationIndicatorNames.get(Concepts.DUPLICATE));
+		desc.setAssociationTargets(Maps.newHashMap("POSSIBLY_EQUIVALENT_TO", Sets.newHashSet(anotherConceptId)));
+		concept.setInactivationIndicator(Concepts.inactivationIndicatorNames.get(DUPLICATE));
+		transferredConcept = simulateRestTransfer(concept);
+		conceptService.update(transferredConcept, mainBranchPath);
+		concept = conceptService.find(concept.getConceptId(), mainBranchPath);
+
+		// Create next version
+		codeSystemService.createVersion(codeSystem, 20210531, "20210531");
+
+		// Create an extension code system (Sweden)
+		codeSystemService.createCodeSystem(new CodeSystem("SNOMEDCT-SE", extensionBranchPath));
+		branchService.updateMetadata(extensionBranchPath, ImmutableMap.of(Config.DEFAULT_MODULE_ID_KEY, extensionModule, Config.DEFAULT_NAMESPACE_KEY, "1000052"));
+		concept = conceptService.find(concept.getConceptId(), extensionBranchPath);
+
+		assertFalse(concept.isActive());
+		assertEquals(mainBranchPath, concept.getPath());
+
+		// Create an extra description in the extension
+		Description swedishBacon = new Description(swedishFsnTerm).setTypeId(FSN).setCaseSignificance(ci)
+				.setAcceptabilityMap(svPreferred).setActive(true).setLang("sv").setLanguageCode("sv")
+				.addLanguageRefsetMember(swedishLanguageReferenceSet, Concepts.PREFERRED, true);
+		concept.addDescription(swedishBacon);
+		transferredConcept = simulateRestTransfer(concept);
+		conceptService.update(transferredConcept, extensionBranchPath);
+		concept = conceptService.find(concept.getConceptId(), extensionBranchPath);
+
+		// Count: in reference member set, find the module id for the (international) FSN; this should not be the extension module.
+		int countIntlDescrRefSetInExtensionModule =0;
+		descriptionIterator = concept.getDescriptions().iterator();
+		while(descriptionIterator.hasNext()) {
+			Description d = descriptionIterator.next();
+			if (d.getDescriptionId().equals(fsnDesc.getDescriptionId())) {
+				List<ReferenceSetMember> members = referenceSetMemberService.findMembers(extensionBranchPath, d.getDescriptionId(), PageRequest.of(0, 10)).getContent();
+				for (ReferenceSetMember rsm : members) {
+					if (extensionModule.equals(rsm.getModuleId())) {
+						countIntlDescrRefSetInExtensionModule++;
+					}
+				}
+			}
+		}
+		assertEquals(0, countIntlDescrRefSetInExtensionModule);
+
+		// Re-activate the concept
```

Comment:
Well having an extension re-activate a core concept is an interesting case, and one we should probably look at.   But for this test, if you just move this code up to before you start working in the extension branch, then we have a more likely scenario.

If you reactivate in the extension, then ALL the components (except the active descriptions) will jump - quite correctly - into the extension module.    What we're trying to spot here is components doing that when they _shouldn't_.